frontend/packages/knative-plugin/src/components/overview/RevisionsOverviewList.tsx

```diff
@@ -44,34 +48,44 @@ const RevisionsOverviewListItem: React.FC<RevisionsOverviewListItemProps> = ({
   );
 };
 
-const RevisionsOverviewList: React.FC<RevisionsOverviewListProps> = ({ revisions, service }) => (
-  <>
-    <SidebarSectionHeading text="Revisions" className="revision-overview-list">
-      {/* add extra check, if sidebar is opened for a knative deployment */}
-      {service.kind === ServiceModel.kind && (
-        <Button
-          variant="secondry"
-          onClick={() => trafficModalLauncher({ obj: service })}
-          disabled={!(revisions && revisions.length)}
-        >
-          Set Traffic Distribution
-        </Button>
+const RevisionsOverviewList: React.FC<RevisionsOverviewListProps> = ({ revisions, service }) => {
+  const canSetTrafficDistribution = useAccessReview({
+    group: ServiceModel.apiGroup,
+    resource: ServiceModel.plural,
+    namespace: service.metadata.namespace,
+    verb: 'update',
+  });
+
+  return (
+    <>
+      <SidebarSectionHeading text="Revisions" className="revision-overview-list">
+        {/* add extra check, if sidebar is opened for a knative deployment */}
+        {canSetTrafficDistribution &&
+          (service.kind === ServiceModel.kind && (
```

Comment:
I know your didn't add this but why is this here? When will a service not have the right kind? And when do we open this sidebar for a knative deployment?

Reply:
Captured as tech debt item as to not hold this up: https://jira.coreos.com/browse/ODC-2304
cc @invincibleJai